src/rpc/masternode.cpp

```diff
@@ -380,12 +380,11 @@ UniValue masternode_payments(const JSONRPCRequest& request)
     }
 
     int64_t nCount = request.params.size() > 2 ? ParseInt64V(request.params[2], "count") : 1;
-    int nStepsLeft = std::abs(nCount);
 
     // A temporary vector which is used to sort results properly (there is no "reverse" in/for UniValue)
     std::vector<UniValue> vecPayments;
 
-    while (nStepsLeft != 0 && pindex != nullptr) {
+    while (vecPayments.size() < std::abs(nCount) != 0 && pindex != nullptr) {
```

Comment:
why std::abs?

Reply:
`nCount` can be negative

Reply:
Only if the person puts in a negative value right? If so, we should just throw an error if they pass in a negative value

Reply:
https://github.com/dashpay/dash/pull/3863/files#diff-244e87df3ed336aadc56ec7c8fc521954751f3c3f66b3eb3832bcbcc87996c51R335

Reply:
Ahhhhh, coool